code/processes/gateway.q

```diff
@@ -446,7 +449,7 @@ asyncexecjpts:{[query;servertype;joinfunction;postback;timeout;sync]
    :();
    ];
   ];
- query:({[u;q]$[`.pm.execas ~ key `.pm.execas;value (`.pm.execas; q; u);value q]}; .z.u; query);
+ query:({[u;q]$[`.pm.execas ~ key `.pm.execas;value (`.pm.execas; q; u);valp q]}; .z.u; query);
```

Comment:
you need an existence check in here for valp. We have people who use the TorQ gateway standalone, so wouldn't have loaded the permissions lib

Reply:
This is a seperate but identical valp that's defined in the .gw namespace, should I change it to just use the .pm namespace valp and have an existence check?

Reply:
So the "query" function is called on the remote process. It would need to have valp defined. But really it should be an additional conditional, something like 

query:({[u;q]$[`.pm.execas ~ key `.pm.execas;value (`.pm.execas; q; u); `.pm.valp ~ key `.pm.valp; .pm.valp q; value q]}; .z.u; query);